### PR TITLE
feat: setting operation mode

### PR DESCRIPTION
**Pull request by ismarslomic**
- feat: support setting operation modes Control individually (Heating) and Off
- chore: refactor GET and POST request functions and add debug logging
- doc: add short description of the module and link to Mill Rest API doc

### CODE REVIEW

**Comment by ismarslomic** [comment]
@Danielhiversen could you please review this PR so I can proceed with PR on HA Mill Integration with latest py version? Thanks!

File: README.md
Comment (Danielhiversen) [inline README.md]:
```suggestion
```

File: mill_local/__init__.py
Comment (Danielhiversen) [inline mill_local/__init__.py]: Will this fail if `response.status != 200` ?

**Review by Danielhiversen — CHANGES_REQUESTED** [review]

File: mill_local/__init__.py
Comment (ismarslomic) [reply mill_local/__init__.py]: Not 100% sure, but you are right. We should maybe check the value of `response.status` before decoding the `response` to `json`. 

Note that json decoding might fail for other reasons as well, even `response.status `is `200`, and it will throw exception (see [doc](https://docs.aiohttp.org/en/stable/client_quickstart.html#json-response-content)). 

So it might be safer to wrap in a try/except block?

Same guarding should be done in the `_get_request()` function.

The original `_request()` did neither check for `response.status`, but `get_status()` wrapped the function call in a try/except. I think its cleaner to do this checking and try/except wrappings within the `*_request()` functions.

File: README.md
Comment (ismarslomic) [reply README.md]: Why would you remove this clarification?

**Comment by ismarslomic** [comment]
@Danielhiversen I have pushed new changes and fixed PR comments, in addition to few small adjustments to the type hints.

Regarding error handling in the python module, it is not clear for me what strategy is being followed currently in the module. From the HA developer docs I have found following practices:

### Error handling when refreshing data
- `asyncio.TimeoutError` and `aiohttp.ClientError` are already handled by the `data update coordinator`, so we can/should propagate these exceptions from `fetch_heater_and_sensor_data()` in python module, I guess?
- related HA docs: [integration_fetching_data#polling-api-endpoints](https://developers.home-assistant.io/docs/integration_fetching_data#polling-api-endpoints)

### Error handling at integration setup
- in `async_setup_entry` we should raise `ConfigEntryNotReady` for connection errors and HA will do automatic retry. Currently we catch `aiohttp.client_exceptions.ClientError` and `asyncio.TimeoutError` in `get_status()` and return None, maybe we should forward these exceptions to integration?
- we should include error message as first argument in `ConfigEntryNotReady` so HA can include it in the retry logs.
- related HA docs: [integration_setup_failures#handling-offline-or-unavailable-devices-and-services](https://developers.home-assistant.io/docs/integration_setup_failures#handling-offline-or-unavailable-devices-and-services)

### Error handling for changing state commands
- what should we do with errors thrown during state changes, for instance setting target temperature or HVAC mode? Catch and log, but not forward to HA integration? I did not find any recommendations in HA docs about these use cases.

**Comment by Danielhiversen** [comment]
Best would be to raise an exception in this library. For example with `resp.raise_for_status()`
That would require updates in the HA integration.

File: mill_local/__init__.py
Comment (Danielhiversen) [inline mill_local/__init__.py]:
```suggestion
                    _LOGGER.error("POST failed to parse json response", exc_info=True)
```

**Comment by ismarslomic** [comment]
> Best would be to raise an exception in this library. For example with `resp.raise_for_status()`
> That would require updates in the HA integration.

I guess this comment is related to all three scenarios I have mentioned? In that case, we should make sure that all exceptions are being raised in this library. I can do these changes!

I'm working with changes in HA integration in parallell, so I can make sure to catch relevant exceptions and raise ones mentioned in HA doc. However, still not sure what to do with exceptions caught in HA Integrations for state change commands.

File: mill_local/__init__.py
Comment (ismarslomic) [reply mill_local/__init__.py]: Good catch, I forgot to add `command` as argument. Fixed in new commit!

**Comment by Danielhiversen** [comment]
Great !


>  However, still not sure what to do with exceptions caught in HA Integrations for state change commands.

I think any exception is caught and logged. I think it would be fine if pyMillLocal raise an error with an explaining message

**Comment by ismarslomic** [comment]
Im back after longer vacation! Please see my latest commit with following changes:
- raises error (`ClientResponseError`) in all functions
- add type hints to all return values
- add tests and mock HTTP responses (this is valuable as I have plan to suggest other changes to this library in future PRs)

**Comment by ismarslomic** [comment]
I also reached out to the HA core team with questions related to error handling in python libraries for custom integrations. Martin Hjelmare replied following:

> In the coordinator update method you can raise `UpdateFailed` and the coordinator will handle raising `ConfigEntryNotReady` for the first refresh during config entry setup.
In service handlers you should raise `HomeAssistantError` with a message as argument for api errors. The message will be shown in the frontend.
See also https://developers.home-assistant.io/docs/integration_quality_scale_index/#silver-

File: mill_local/__init__.py
Comment (Danielhiversen) [inline mill_local/__init__.py]:
```suggestion
    def __init__(self, device_ip: str, websession: ClientSession, timeout=15: int) -> None:
```

**Comment by ismarslomic** [comment]
@Danielhiversen I fixed your last change request, in addition to linting errors. Added also new step to the code-checker workflow to run tests. Any other input before merging?

**Pull request merged**
